pandas/tests/indexes/test_range.py

```diff
@@ -424,6 +424,14 @@ def test_get_indexer_limit(self):
         expected = np.array([0, 1, 2, 3, 3, -1], dtype=np.intp)
         tm.assert_numpy_array_equal(result, expected)
 
+    @pytest.mark.parametrize("stop", [0, -1, -2])
+    def test_get_indexer_decreasing(self, stop):
+        # GH 28678
+        index = RangeIndex(7, stop, -3)
+        result = index.get_indexer(range(9))
+        expected = np.array([-1, 2, -1, -1, 1, -1, -1, 0, -1], dtype=np.intp)
```

Comment:
Is there a reason to have to specify `np.intp` here?

Reply:
`intp` is the native numpy indexing type (see first bullet point [here](https://docs.scipy.org/doc/numpy-1.17.0/reference/arrays.indexing.html#detailed-notes) and [this table](https://docs.scipy.org/doc/numpy-1.10.1/user/basics.types.html#array-types-and-conversions-between-types)) so we typically match that in our indexing code